internal/subtitles/client.go

```
@@ -0,0 +1,221 @@
+
+	// The link is unauthenticated and comes straight from the API response,
+	// so validate it before following: require an absolute https URL. The
+	// one exception is a loopback host over http, which only exists so
+	// httptest-backed tests (that serve both the API and the "CDN" from the
+	// same local server) keep working — a real download link must be https.
+	link, err := url.Parse(parsed.Link)
+	if err != nil || link.Host == "" || !(link.Scheme == "https" || (link.Scheme == "http" && isLoopbackHost(link.Hostname()))) {
+		return nil, fmt.Errorf("subtitles: download link is not an absolute https URL: %q", parsed.Link)
+	}
+
+	fileResp, err := c.httpClient().Get(link.String())
```

golangci-lint (2.12.2)

[error] (*net/http.Client).Get must not be called. use (*net/http.Client).Do(*http.Request)

(noctx)